Pal 291 jvm launcher logging switch (#68)
gchq/Palisade-services: services-manager/README.md

## Getting started

The Palisade Services Manager is a configuration-driven process-spawner and REST-client with particular considerations to assist in managing multiple Palisade services running on local JVMs.

The manager is designed to be used by defining a collection of SpringBoot configuration files:
 * One set for the manager itself - this defines how it should start up further services or reconfigure those already running
 * One set for each service - these are discovered through the above configuration and passed on as an argument to the service jar
    * These wll all already be located in each service's `resources` directory
 
 The manager works in several modes, operated by the use of several flags:
 * `--run` - run the configured services, spawning a new process for each entry
 * `--logging` - perform a change to the logging level of configured services by REST requests
 * `--config` - print out a human-readable view of the spring-boot configuration given to the services-manager
 
 
## Examples

### Starting Services
If services are already running, using the built-in profiles:  

>  1. Start Eureka - run `java -jar -Dspring.profiles.active=eureka services-manager-exec.jar --run` and wait for Eureka to start up on `localhost:8083`

Reviewer: there is no jar called services-manager-exec.jar. put in the path from the project root and use a maven plugin to embed the current build jar name into the readme. People need to be able to just copy-pate these into the window.

Author: Resolved with maven-resources-plugin/copy-resources and maven-replacer-plugin/replace

<!---

> Copyright 2019 Crown Copyright

Reviewer 2: ```suggestion
Copyright 2020 Crown Copyright
```

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
--->

<!---

> This file is under substitution in the build process - maven's `process-resources` stage will substitute executable(dot)jar for services-manager-0.4.0-SNAPSHOT-exec.jar

Reviewer 2: ```suggestion
This file is under substitution in the build process - maven's `process-resources` stage will substitute executable(dot)jar for services-manager-<version>-exec.jar
```

Reviewer 2: Given this statement then I believe all the occurrences of services-manager-0.4.0-SNAPSHOT-exec.jar should actually be executable(dot)jar. To prevent this happening again it might be better to have a template readme and then a generated readme which is in the git ignore.

Author: Will move README.md in resources/doc to README_TEMPLATE.md to clarify behaviour

    * These wll all already be located in each service's `resources` directory
 
 The manager works in several modes, operated by the use of several flags:
 * `--run` - run the configured services, spawning a new process for each entry
 * `--logging` - perform a change to the logging level of configured services by REST requests
 * `--config` - print out a human-readable view of the spring-boot configuration given to the services-manager
 
 
## Examples

### Starting Services
If services are already running, using the built-in profiles:  
 1. Start Eureka - `java -jar -Dspring.profiles.active=eureka services-manager-0.4.0-SNAPSHOT-exec.jar --run` 
    * Wait for Eureka to start up on `localhost:8083`
 3. Start all other services - `java -jar -Dspring.profiles.active=services services-manager-0.4.0-SNAPSHOT-exec.jar --run` 
    * Services should now begin registering with Eureka and appear on the dashboard
 4. Begin using Palisade (see [palisade-examples](https://github.com/gchq/Palisade-examples))  

### Enabling Debug Logging

#### At Start-Time
If services are not running, or debug logging is required from startup, using the built-in profiles:  
 1. Start Eureka as above

>  2. Add the `debug` profile to the services runner - `java -jar -Dspring.profiles.active=services,debug services-manager-0.4.0-SNAPSHOT-exec.jar --run` 

Reviewer 2: ```suggestion
 2. Add the `debug` profile to the services runner - `java -jar -Dspring.profiles.active=services,debug services-manager-*-exec.jar --run` 
```

Reviewer 2: Or we could just do it this way and remove the substitution stage
